massa-execution-worker/src/interface_impl.rs

```diff
@@ -603,6 +603,49 @@ impl Interface for InterfaceImpl {
         Ok(public_key.verify_signature(&h, &signature).is_ok())
     }
 
+    /// Verify an EVM signature
+    ///
+    /// Information:
+    /// * Expects a SECP256K1 signature in full ETH format.
+    ///   Format: (r, s, v) v will be ignored
+    ///   Length: 65 bytes
+    /// * Expects a public key in full ETH format.
+    ///   Length: 65 bytes
+    fn verify_evm_signature(
+        &self,
+        signature_: &[u8],
+        message_: &[u8],
+        public_key_: &[u8],
+    ) -> Result<bool> {
+        // parse the public key
+        let public_key = libsecp256k1::PublicKey::parse_slice(
+            public_key_,
+            Some(libsecp256k1::PublicKeyFormat::Full),
+        )?;
+
+        // build the message
+        let prefix = format!("\x19Ethereum Signed Message:\n{}", message_.len());
+        let to_hash = [prefix.as_bytes(), message_].concat();
+        let full_hash = sha3::Keccak256::digest(to_hash);
+        let message = libsecp256k1::Message::parse_slice(&full_hash).unwrap();
+
+        // parse the signature as being (r, s, v)
+        // r is the R.x value of the signature's R point (32 bytes)
+        // s is the signature proof for R.x (32 bytes)
+        // v is a recovery parameter used to ease the signature verification (1 byte)
+        // we ignore the recovery parameter here
+        // see test_evm_verify for an example of its usage
+        let signature = libsecp256k1::Signature::parse_standard_slice(&signature_[..64]).unwrap();
```

Comment:
the unwrap here can be triggered by an attacker sending an invalid signature

Reply:
fixed